chore(mcp): source a gitignored .env in ensure-tunnel.sh when present

## Summary

Make `scripts/ensure-tunnel.sh` source a project-root `.env` (when present) before it launches the tunnel, so harness config (`GEMINI_API_KEY`, `AETHER_GEMINI_*`, etc.) flows into the tunnel → hub → spawned substrates via the existing env-inheritance chain — without the operator manually exporting vars first. No-op when the file is absent.

`.env` is gitignored as of #1237 (covers `.env`, `.env.*`, `*.env`), so sourcing a developer-local `.env` is safe and the file is never committed.

## Why

Harness config enters the process tree only via the shell that launches the tunnel: the tunnel forks the hub + mcp inheriting its environment, and substrates inherit the hub's (no `env_clear`). Today operators must remember to export vars before running `ensure-tunnel.sh`; this wires the missing convenience.

## Change

In `scripts/ensure-tunnel.sh`, **after `PROJECT_DIR` is resolved** (the `PROJECT_DIR="${CLAUDE_PROJECT_DIR:-}"` + fallback block) and **before the tunnel launch**, add:

```bash
# Load a developer-local .env if present (gitignored — never committed). Auto-export
# so the tunnel and everything it forks inherit it. No-op when the file is absent.
if [[ -f "$PROJECT_DIR/.env" ]]; then
    set -a
    # shellcheck disable=SC1091
    . "$PROJECT_DIR/.env"
    set +a
fi
```

## CRITICAL constraints (security)
- This PR adds **only** sourcing logic. It must **NOT** create, add, commit, or otherwise include any `.env` file in the diff.
- Do not modify the `.env` entries in `.gitignore`.
- Do not echo or print `.env` contents anywhere in the script.
- When `$PROJECT_DIR/.env` is absent, the script must behave exactly as before (the `[[ -f ]]` guard makes it a no-op).

## Validation
- `git status` shows only `scripts/ensure-tunnel.sh` changed; no `.env` added; `git ls-files` shows no tracked `.env`.
- `shellcheck scripts/ensure-tunnel.sh` is clean (the `SC1091` disable covers the dynamic source).
- With no `.env` present, `bash scripts/ensure-tunnel.sh` behaves identically to before.
- Scripts-only change → CI/repo-config class, no Rust pre-flight required.

--- a/scripts/ensure-tunnel.sh
+++ b/scripts/ensure-tunnel.sh
@@ -38,6 +38,15 @@ if [[ -z "$PROJECT_DIR" ]]; then
     PROJECT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
 fi
 
+# Load a developer-local .env if present (gitignored — never committed). Auto-export
+# so the tunnel and everything it forks inherit it. No-op when the file is absent.
+if [[ -f "$PROJECT_DIR/.env" ]]; then
+    set -a
+    # shellcheck disable=SC1091
+    . "$PROJECT_DIR/.env"
+    set +a
+fi
+
 # Probe :8890 — true if the tunnel is answering. Prefers the /admin/status
 # HTTP probe (curl), falls back to a bare TCP connect (nc). Both swallow
 # their own failure so the caller controls the exit.

